ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/recovery/recoverycoordinators/GenericRecoveryCoordinator.java

```diff
@@ -226,6 +229,14 @@ protected static Status replay_completion ( RecoveryCoordinatorId id, Resource r
 
 	    String tranType = ( (id._isServerTransaction) ? ServerTransaction.typeName() : ArjunaTransactionImple.typeName() );
 
+        try {
+            if (id._isServerTransaction && (StoreManager.getRecoveryStore().currentState(id._actionUid, ServerTransaction.typeName() + "/JCA") != StateStatus.OS_UNKNOWN)) {
```

Comment:
Just a comment here but it is a shame we must hard code knowledge of the opaque typeName of the jtax ServerTransaction. Other than providing a type name repository I do not see how to avoid it.